status-im/status-legacy
# Fix/8702 hash passwords

### Summary

fix #8702

clean up native-module

- remove impl/modules.cljs namespace because it was needlessly redundant
- remove `create-account` unused call
- add documentation for some calls
- only keep `(status)` check for relevant calls that are failing tests
because they are evaluated

- use ethereum.utils sha3 function to hash passwords before sending them
to status-go
- some native calls take password as one of their params: they now take
hashed-password (still a string, only relevant for status-react caller)
- some native calls take password within a map of rpc params: it now
needs to be hashed with `ethereum.core/sha3`

#### Areas that maybe impacted
Any area that takes a password as input

### Steps to test

Test 
- login
- recovery
- adding account
- signing message
- signing transaction

changes are not platform dependent so it is not a big deal if iOS doesn't build
(I tested all of these on Android emulator on Ropsten with simpledapp.eth)

No password should be logged

status: ready
